agent/engine/daemonmanager/daemon_manager_linux.go

```diff
@@ -51,157 +29,3 @@ const (
 	socketPathHostRoot                          = "/var/run/ecs"
 	logPathHostRoot                             = "/log/daemons"
 )
-
-var mkdirAllAndChown = utils.MkdirAllAndChown
-
-func (dm *daemonManager) CreateDaemonTask() (*apitask.Task, error) {
-	imageName := dm.managedDaemon.GetImageName()
-	// create daemon-specific host mounts
-	if err := dm.initDaemonDirectoryMounts(imageName); err != nil {
-		logger.Error("initDaemonDirectory failure",
-			logger.Fields{
-				field.Error: err,
-			})
-		return nil, err
-	}
-
-	loadedImageRef := dm.managedDaemon.GetLoadedDaemonImageRef()
-	containerRunning := apicontainerstatus.ContainerRunning
-	stringCaps := []string{}
-	if dm.managedDaemon.GetLinuxParameters() != nil {
-		caps := dm.managedDaemon.GetLinuxParameters().Capabilities.Add
-		for _, cap := range caps {
-			stringCaps = append(stringCaps, *cap)
-		}
-	}
-	dockerHostConfig := dockercontainer.HostConfig{
-		Mounts:      []dockermount.Mount{},
-		NetworkMode: apitask.HostNetworkMode,
-		// the default value of 0 for MaximumRetryCount means retry indefinitely
-		RestartPolicy: dockercontainer.RestartPolicy{
-			Name:              "on-failure",
-			MaximumRetryCount: 0,
-		},
-		Privileged: dm.managedDaemon.GetPrivileged(),
-		CapAdd:     stringCaps,
-	}
-	if !dm.managedDaemon.IsValidManagedDaemon() {
-		return nil, fmt.Errorf("%s is an invalid managed daemon", imageName)
-	}
-
-	for _, mp := range dm.managedDaemon.GetMountPoints() {
-		if err := mkdirAllAndChown(mp.SourceVolumeHostPath, daemonMountPermission, daemonUID, os.Getegid()); err != nil {
-			return nil, err
-		}
-		var bindOptions = dockermount.BindOptions{}
-
-		if mp.PropagationShared {
-			// https://github.com/moby/moby/blob/master/api/types/mount/mount.go#L52
-			bindOptions = dockermount.BindOptions{Propagation: dockermount.PropagationShared}
-		}
-		logger.Info(fmt.Sprintf("bindMount Propagation: %s", bindOptions.Propagation),
-			logger.Fields{
-				field.Image: loadedImageRef,
-			})
-		mountPoint := dockermount.Mount{
```

Comment:
Why mounts were skipped in the common code?

Reply:
No mounts were skipped in the common code. The previous code was hardcoding the mount type to `bind` and we want some of our mounts to be `npipe` which we are passing from upstream and processing accordingly. 

https://github.com/saurabhc123/amazon-ecs-agent/pull/2/files#diff-c27a802d01286c3a45f84d314f4bf1feb2448370cba4991ebf67d9afdc433d5aR116-R131